packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx

```diff
@@ -45,8 +47,21 @@ export function TreeRenderer({
       return (
         <TreeNodeRenderer
           {...nodeProps}
+          onNodeClick={(nodeId, isSelected, event) => {
```

Comment:
Maybe we should handle double clicks in `onNodeClick` as now we have code related to double clicks in both events?

Reply:
Moved into `onNodeClick`.